sys/include/net/netstats.h

```diff
@@ -53,6 +75,45 @@ typedef struct {
     uint32_t rx_bytes;          /**< received bytes */
 } netstats_t;
 
+/**
+ * @brief       Stats per peer struct
+ */
+typedef struct {
+    uint8_t  l2_addr[CONFIG_L2ADDR_MAX_LEN]; /**< Link layer address of the neighbor */
+    uint8_t  l2_addr_len;   /**< Length of netstats_nb::l2_addr */
+    uint8_t  freshness;     /**< Freshness counter */
+#ifdef MODULE_NETSTATS_NEIGHBOR_ETX
+    uint16_t etx;           /**< ETX of this peer */
+#endif
+#ifdef MODULE_NETSTATS_NEIGHBOR_TX_TIME
+    uint32_t time_tx_avg;   /**< Average frame TX time in µs */
+#endif
+#ifdef MODULE_NETSTATS_NEIGHBOR_RSSI
+    uint8_t  rssi;          /**< Average RSSI of received frames in abs([dBm]) */
+#endif
+#ifdef MODULE_NETSTATS_NEIGHBOR_LQI
+    uint8_t  lqi;           /**< Average LQI of received frames */
+#endif
+#ifdef MODULE_NETSTATS_NEIGHBOR_COUNT
+    uint16_t tx_count;      /**< Number of sent frames to this peer */
+    uint16_t tx_fail;       /**< Number of sent frames that did not get ACKed */
+    uint16_t rx_count;      /**< Number of received frames */
+#endif
+    uint16_t last_updated;  /**< seconds timestamp of last update */
+    uint16_t last_halved;   /**< seconds timestamp of last halving */
+} netstats_nb_t;
+
+/**
+ * @brief       L2 Peer Info struct
+ */
+typedef struct {
+    cib_t stats_idx;                                        /**< CIB for the tx correlation     */
+    netstats_nb_t *stats_queue[NETSTATS_NB_QUEUE_SIZE];     /**< send/callback mac association array */
```

Comment:
This comment needs to be added on top, I think. Otherwise, you exceed even the soft limit on line length here :-)'